src/main/java/at/hannibal2/skyhanni/api/SkillAPI.kt

```diff
@@ -0,0 +1,237 @@
+package at.hannibal2.skyhanni.api
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.ConfigLoadEvent
+import at.hannibal2.skyhanni.events.LorenzActionBarEvent
+import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.TabListData
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import com.google.common.base.Splitter
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonArray
+import com.google.gson.reflect.TypeToken
+import io.github.moulberry.notenoughupdates.util.Constants
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.init.Blocks
+import net.minecraft.init.Items
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.io.Serializable
+import java.util.regex.Matcher
+
+object SkillAPI {
+    private val patternGroup = RepoPattern.group("display.skilldisplay")
+    private val SKILL_PATTERN_PERCENT by patternGroup.pattern("skillpaternpercent", "\\+(\\d+(?:,\\d+)*(?:\\.\\d+)?) (.+) \\((\\d\\d?(?:\\.\\d\\d?)?)%\\)")
```

Comment:
The patterns are unnecessarily unreadable.
I won't go over each one, but takes this one as example to write them better.
```suggestion
    private val SKILL_PATTERN_PERCENT by patternGroup.pattern("skillpaternpercent", "\\+[\\d.,]+ (.+) \\(([\\d.]{1,5})%\\))")
```